pkg/cli/cmds/agent.go

```diff
@@ -9,6 +9,13 @@ import (
 )
 
 var (
+	agentFlag = []cli.Flag{
+		&cli.StringFlag{
+			Name:   "cni",
+			Usage:  "(networking/windows) CNI Plugins to configure on windows only calico and none are supported (default: calico)",
+			EnvVar: "RKE2_CNI",
+		},
+	}
```

Comment:
This means we are adding the `--cni` flag in the agent and it will appear on both linux and windows, which might be a bit confusing for the user. The only reason to have this is to support `cni: none` only on the windows node, right? Is there a use case for that?

Reply:
Before it was selecting `cni: none` based on what helm chart it detected. It was not possible to have a different cni plugin only on windows agent:
```
func getCniType(restConfig *rest.Config) (string, error) {
	hc, err := helm.NewFactoryFromConfig(restConfig)
	if err != nil {
		return "", err
	}
	hl, err := hc.Helm().V1().HelmChart().List(metav1.NamespaceSystem, metav1.ListOptions{})
	if err != nil {
		return "", err
	}
	for _, h := range hl.Items {
		if h.Name == win.CalicoChart {
			return "calico", nil
		} else if h.Name == "rke2-cilium" {
			return "cilium", nil
		} else if h.Name == "rke2-canal" {
			return "canal", nil
		}
	}
	return "none", nil
}
```

Reply:
And then:
```
	cniType, err := getCniType(restConfig)
	if err != nil {
		return err
	}

	switch cniType {
	case "calico":
		p.cni = &win.Calico{}
	case "none":
		logrus.Info("CNI is set to none, skipping CNI configuration")
		return nil
	default:
		return fmt.Errorf("the CNI %s isn't supported on Windows", cniType)
	}
```

Reply:
yes but before was using the cni config of the server but reading from the issue and the discussion it seems like that the desired behaviour was to explicitly disable the CNI on windows but not on the Linux nodes.

Reply:
I moved it to be shown only on the windows agent

Reply:
ok